main/src/propagator/factory.hpp

```diff
@@ -46,6 +46,31 @@ template<class DomainType, class ParticleDataType>
 std::unique_ptr<Propagator<DomainType, ParticleDataType>>
 propagatorFactory(const std::string& choice, size_t ngmax, size_t ng0, std::ostream& output, size_t rank)
 {
+#ifdef USE_HIP_VE
```

Comment:
Since only one of them will be used, this part should match the CMake changes using: #ifdef USE_HIP_STD ... #elif USE_HIP_TURBULENCE ... #elif USE_HIP_VE. I also think we should add another wrapper #if USE_HIP or something like this and keep the CUDA version with another #ifdef USE_CUDA

Reply:
I said CUDA but it should be there for non GPU code as well.

Reply:
I suggest the following:
#ifdef USE_HIP

#ifdef USE_HIP_STD
    if (choice == "std")
    {
        return std::make_unique<HydroProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else { throw std::runtime_error("Unknown propagator choice (HIP): " + choice); }
#endif
#ifdef USE_HIP_TURBULENCE
    if (choice == "turbulence")
    {
        return std::make_unique<TurbVeProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else { throw std::runtime_error("Unknown propagator choice (HIP): " + choice); }
#endif
#ifdef USE_HIP_VE
    if (choice == "ve")
    {
        return std::make_unique<HydroVeProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else { throw std::runtime_error("Unknown propagator choice (HIP): " + choice); }
#endif // USE_HIP_STD or TURBULENCE or VE

#else
    if (choice == "ve")
    {
        return std::make_unique<HydroVeProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else if (choice == "std")
    {
        return std::make_unique<HydroProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else if (choice == "turbulence")
    {
        return std::make_unique<TurbVeProp<DomainType, ParticleDataType>>(ngmax, ng0, output, rank);
    }
    else { throw std::runtime_error("Unknown propagator choice: " + choice); }
}
#endif // USE_HIP